File: algorithms/1_easy/S/02553_separate_the_digits_in_an_array/solution.py
class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        res = []

        for num in nums:
            p = 1
            while p <= num:
                p *= 10

            p //= 10

            while p > 0:
                res.append((num // p) % 10)
                p //= 10

        return res

File: algorithms/1_easy/S/02553_separate_the_digits_in_an_array/solution.rs
impl Solution {
    pub fn separate_digits(nums: Vec<i32>) -> Vec<i32> {
        let mut res: Vec<i32> = Vec::new();

        for num in nums.into_iter() {
            let mut p = 1_i32;
            while p <= num {
                p *= 10;
            }
            p /= 10;

            while p > 0 {
                res.push((num / p) % 10);
                p /= 10;
            }
        }

        res
    }
}
